app/src/main/java/com/movie/app/details/DetailsMovieActivity.kt

```diff
@@ -4,27 +4,28 @@ import android.content.Context
 import android.content.Intent
 import android.graphics.Color
 import android.os.Bundle
-import android.support.v7.widget.LinearLayoutManager
 import android.view.View
+import androidx.lifecycle.Observer
+import androidx.recyclerview.widget.LinearLayoutManager
 import com.movie.app.BaseActivity
 import com.movie.app.R
 import com.movie.app.modules.Movie
 import com.movie.app.util.GenreUtil
+import com.movie.app.util.Result
 import com.movie.app.util.enableToolbarBack
 import com.movie.app.util.loadImage
 import com.movie.app.util.setToolbar
 import com.movie.app.util.setToolbarTitle
 import com.pierfrancescosoffritti.youtubeplayer.player.AbstractYouTubePlayerListener
 import kotlinx.android.synthetic.main.activity_details_movie.*
-import javax.inject.Inject
+import org.koin.androidx.viewmodel.ext.android.viewModel
 
-class DetailsMovieActivity : BaseActivity(), DetailsActivityContractor.View {
+class DetailsMovieActivity : BaseActivity() {
 
-    @Inject
-    lateinit var presenter: DetailsActivityContractor.Presenter
+    private val viewModel: DetailsMovieViewModel by viewModel()
```

Comment:
**Suggestion:** Consider using a ViewModelFactory for injecting the ViewModel, instead of directly using the 'by viewModel()' delegation. This would make the code more testable and decoupled.
```suggestion
    private val viewModelFactory: ViewModelProvider.Factory
    private val viewModel: DetailsMovieViewModel by viewModels { viewModelFactory }
```